docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1135 @@
 
-## Product scope
-### Target user profile
+[&#8593; Return to Top](#developer-guide)
 
-{Describe the target user profile}
+## 2. <a id="setting-up">Setting Up</a>
 
-### Value proposition
+### 2.1. <a id="prerequisites">Prerequisites</a>
+1. JDK `11`.
+2. IntelliJ IDEA IDE.
 
-{Describe the value proposition: what problem does it solve?}
+### 2.2. <a id="setting-up-the-project-in-your-computer">Setting up the Project in Your Computer</a>
+1. Fork [this repository](https://github.com/AY2021S1-CS2113T-F11-1/tp), and clone the fork to your computer.
+2. Open IntelliJ (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle  
+   1. Click `Configure` > `Structure for New Projects` and then `Project Settings` > `Project` > `Project SDK`.
+   2. If `JDK 11` is listed in the drop down, select it. Otherwise, click `New…` and select the directory where you installed `JDK 11`.
+   3. Click `OK`.
+4. Click `Import Project`.
+5. Locate the `build.gradle` file and select it. Click `OK`.
+6. Click `Open as Project`.
+7. Click `OK` to accept the default settings if prompted. 
 
-## User Stories
+[&#8593; Return to Top](#developer-guide)
+
+## 3. <a id="design">Design</a>
+This section provides a high level overview of our application, The Schwarzenegger.
+### 3.1. <a id="architect">Architecture</a>
+
+![Architecture](pictures/zesong/Architecture.png)
+
+The image above explains the design of the application, The Schwarzenegger. 
+
+The main driver of the application is `Main: Duke`. It is responsible for mainly two phases:
+- At application launch
+    - This class will initialise the components in the correct sequence and is in charge of connecting them with each other.
+- At shut down
+    - This class will invoke cleanup method for the components when necessary.
+    
+In addition to that, the architecture of The Schwarzenegger is broken down into several packages, mainly the following: 
+- `Ui`: This class mainly handles the interactions with user of the application.
+- `Parser`: This class mainly handles the parsing and handling of user commands.
+- `Command`: This class handles the type of command.
+- `Profile`: This class manages the data of the user.
+- `Diet`: This class manages the diet recording sessions.
+- `Workout`: This class manages the data workout recording sessions.
+- `Storage`: This class reads data from and writes data back into a text file for future uses.
+
+[&#8593; Return to Top](#developer-guide)
+
+### 3.2. <a id="ui-component">Ui Component</a>
+![Ui Component](images/UI_Component.png)
+
+API: `Ui.java`
+ 
+The `Ui` package is a combination class where all interaction will be made through this component
+ 
+The `Ui` component,
+
+* Takes in user input
+* Prints out response messages
+
+[&#8593; Return to Top](#developer-guide)
+
+### 3.3. <a id="logic-component">Logic Component</a>
+
+![Logic Component](images/logic_update.png)
```

Comment:
Same as the picture of Ui Component, the picture of Logic Component does not show. Would it be better to upload it?
<img width="191" alt="5D9DD571-67D6-4B98-A627-AF1A48E7E88F" src="https://user-images.githubusercontent.com/57485187/97533845-f1d5cc80-19f3-11eb-9675-4089f5a2f5fc.png">